dataikuapi/dss/apideployer.py

```diff
@@ -334,15 +334,29 @@ def start_update(self):
 
         return DSSFuture(self.client, future_response.get('jobId', None), future_response)
 
-    def delete(self):
+    def delete(self, force=False):
         """
-        Deletes this deployment
+        Deletes this deployment.
+
+        :param boolean force: If True, automatically disables deployment before deletion
 
         You may only delete a deployment if it is disabled and has been updated after disabling it.
         """
-        return self.client._perform_empty(
-            "DELETE", "/api-deployer/deployments/%s" % (self.deployment_id))
 
+        # Check is the deployment is disabled
+        is_enabled = self.get_status().light_status.get("deploymentBasicInfo").get("enabled")
+        if (is_enabled and force) or (not is_enabled):
+            if force:
+                settings = self.get_settings()
+                settings.set_enabled(enabled=False)
+                settings.save()
+            return self.client._perform_empty(
+                "DELETE", "/api-deployer/deployments/%s" % (self.deployment_id))
+        else:
+            raise Exception("Deployment {} deletion failed: deployment must be disabled first.".format(self.deployment_id))
```

Comment:
In the current state of the code, we always do a call to `save` when `force=True`, even if deployment is disabled. We could ensure to only disable when needed?

```suggestion
        is_enabled = self.get_status().light_status["deploymentBasicInfo"].get("enabled")
        if is_enabled and not force:
            raise Exception("Deployment {} deletion failed: deployment must be disabled first.".format(self.deployment_id))
        if is_enabled:
            settings = self.get_settings()
            settings.set_enabled(enabled=False)
            settings.save()
        self.client._perform_empty(
                "DELETE", "/api-deployer/deployments/%s" % (self.deployment_id))
```